docs/design/values.md

```diff
@@ -385,15 +381,17 @@ set of heuristics. Some examples:
 -   Small objects that are trivially copied in a machine register would use
     `const Self`.
 
-When a custom type is provided, it must not be `Self` (or the other two cases).
-The type provided will be used on function call boundaries and as the
-implementation representation for `let` bindings and other value expressions
-referencing an object of the type. A specifier of `value_rep = T;` will require
-that the type `impls ReferenceImplicitAs(T)` using the following interface:
+When a custom type is provided, it must not be `Self`, `const Self`, or a
+pointer to either. The type provided will be used on function call boundaries
+and as the implementation representation for `let` bindings and other value
+expressions referencing an object of the type. A specifier of `value_rep = T;`
+will require that the type containing that specifier satisfies the constraint
+`impls ReferenceImplicitAs where .T = T` using the following interface:
 
 ```carbon
-interface ReferenceImplicitAs(T:! type) {
-  fn Op[addr self: const Self*]() -> T;
+interface ReferenceImplicitAs {
+  let T:! type;
```

Comment:
Now that this is an associated type, do we want to say that this interface is implemented for all types, so you can always get this type? For example, we may want to return the value representation type of the element type when iterating through a container.

Reply:
Based on discussion: we don't want to make the `Convert` function available for all types, even if its okay to make the associated type available. This suggests an alternative implementation:
```
interface ValueRepresentation {
  let T:! type;
}

interface ReferenceImplicitAs {
  // Possibly with a template constraint that `T` is not
  // `Self`, `const Self`, `Self*`, or `const Self*`
  extend ValueRepresentation;
  fn Convert...;
}
```
and `ValueRepresentation` would be implemented for all types.

Reply:
Let me know when you've filed the issue for leads.

Reply:
Filed here: https://github.com/carbon-language/carbon-lang/issues/2990